src/Token/AccessToken.php

```diff
@@ -169,6 +169,14 @@ public function getRefreshToken()
         return $this->refreshToken;
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function setRefreshToken($refreshToken)
```

Comment:
Since this isn't part of any interface, if the calling code has type-hinted for the `AccessTokenInterface` or the `ResourceOwnerAccessTokenInterface`, then there's no guarantee that the access token object will have this method.

Rather than changing the existing interfaces, which would break BC, I propose introducing a new interface. Something like `SettableRefreshTokenInterface`. It would have a simple API:

```php
namespace League\OAuth2\Client\Token;

interface SettableRefreshTokenInterface
{
    /**
     * Sets or replaces the refresh token with the provided refresh token
     *
     * @param string $refreshToken
     *
     * @return void
     */
    public function setRefreshToken($refreshToken);
}
```

Then, on `AccessToken`, we would implement this interface.

If anyone wants to hint for both `AccessTokenInterface` and `SettableRefreshTokenInterface`, they may use intersection types (i.e., `AccessTokenInterface & SettableRefreshTokenInterface`) either in native code (PHP 8.1+) or in docblock annotations (for PHPStan, Psalm, etc.).

Thoughts?

Reply:
Thanks for your comment @ramsey.  Good point, I hadn't considered the effects of this method not being defined in any interface.

Creating a new `SettableRefreshTokenInterface` interface like you suggest sounds like the best bet to avoid breaking changes.  Keeping it in `AccessTokenInterface` _would_ be nice since `getRefreshToken()` is already defined there, but maybe something that can be consolidated in the next major release?

I've just committed that new `SettableRefreshTokenInterface` ✅

Reply:
Thanks!